File: 13 Roman to Integer/Python/Solution.py
from typing import Dict, Self


class Solution:
    def romanToInt(self: Self, s: str) -> int:
        res: int = 0

        if "IV" in s or "IX" in s:
            res -= 2
        if "XL" in s or "XC" in s:
            res -= 20
        if "CD" in s or "CM" in s:
            res -= 200
        
        romanToInt: Dict[str, int] = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000
        }
        for char in s:
            res += romanToInt[char]

        return res

File: 13 Roman to Integer/Java/Solution.java
import java.util.Map;
import java.nio.charset.CharacterCodingException;
import java.util.HashMap;


class Solution {
    public int romanToInt(String s) {
        int res = 0;

        if (s.contains("IV") || s.contains("IX")) {
            res -= 2;
        }
        if (s.contains("XL") || s.contains("XC")) {
            res -= 20;
        }
        if (s.contains("CD") || s.contains("CM")) {
            res -= 200;
        }

        Map<Character, Integer> map = new HashMap<>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);
        
        for (int i = 0; i < s.length(); i++) {
            res += map.get(s.charAt(i));
        }

        return res;
    }
}